### PR TITLE
[KIECLOUD-97] - Propagate OpenShiftStartupStrategy related changes to…

### PR DESCRIPTION
**Pull request by spolti**
… KIE APB

Signed-off-by: Filippe Spolti <fspolti@redhat.com>

### CODE REVIEW

**Comment by rhtevan** [comment]
It looks good to me in terms of propagating the latest changes, i.e. adding the lifecycle hooks etc. However,  there are few things missing from APB in comparison to the latest 7.4 Templates  
1. KIE Server Service requires a session affinity configuration, such as this (https://github.com/jboss-container-images/rhpam-7-openshift-image/blob/3160e8313bbaa8c3777a2ac4fa5596e1849e7bf0/templates/rhpam74-kieserver-postgresql.yaml#L424-L428)
2. If KIE Server config a HTTP (Not HTTPs) Route, it needs this (https://github.com/jboss-container-images/rhpam-7-openshift-image/blob/3160e8313bbaa8c3777a2ac4fa5596e1849e7bf0/templates/rhpam74-kieserver-postgresql.yaml#L481)
3. **(IMPORTANT)** For all KIE Server DC, it requires this label at 2 locations. i.e. (https://github.com/jboss-container-images/rhpam-7-openshift-image/blob/3160e8313bbaa8c3777a2ac4fa5596e1849e7bf0/templates/rhpam74-kieserver-postgresql.yaml#L513) and (https://github.com/jboss-container-images/rhpam-7-openshift-image/blob/3160e8313bbaa8c3777a2ac4fa5596e1849e7bf0/templates/rhpam74-kieserver-postgresql.yaml#L544)
4. For all KIE Server DC, it requires this new field: (https://github.com/jboss-container-images/rhpam-7-openshift-image/blob/3160e8313bbaa8c3777a2ac4fa5596e1849e7bf0/templates/rhpam74-kieserver-postgresql.yaml#L517)
5. **(IMPORTANT)** For all KIE Server DC, it requires this new downwardAPI mapping: (https://github.com/jboss-container-images/rhpam-7-openshift-image/blob/3160e8313bbaa8c3777a2ac4fa5596e1849e7bf0/templates/rhpam74-kieserver-postgresql.yaml#L618-L621)

**Comment by rhtevan** [comment]
@ruromero Could you help double check see if there were anything missing. Thanks!

**Comment by spolti** [comment]
> It looks good to me in terms of propagating the latest changes, i.e. adding the lifecycle hooks etc. However, there are few things missing from APB in comparison to the latest 7.4 Templates
> 
> 1. KIE Server Service requires a session affinity configuration, such as this (https://github.com/jboss-container-images/rhpam-7-openshift-image/blob/3160e8313bbaa8c3777a2ac4fa5596e1849e7bf0/templates/rhpam74-kieserver-postgresql.yaml#L424-L428)
> 2. If KIE Server config a HTTP (Not HTTPs) Route, it needs this (https://github.com/jboss-container-images/rhpam-7-openshift-image/blob/3160e8313bbaa8c3777a2ac4fa5596e1849e7bf0/templates/rhpam74-kieserver-postgresql.yaml#L481)
> 3. **(IMPORTANT)** For all KIE Server DC, it requires this label at 2 locations. i.e. (https://github.com/jboss-container-images/rhpam-7-openshift-image/blob/3160e8313bbaa8c3777a2ac4fa5596e1849e7bf0/templates/rhpam74-kieserver-postgresql.yaml#L513) and (https://github.com/jboss-container-images/rhpam-7-openshift-image/blob/3160e8313bbaa8c3777a2ac4fa5596e1849e7bf0/templates/rhpam74-kieserver-postgresql.yaml#L544)
> 4. For all KIE Server DC, it requires this new field: (https://github.com/jboss-container-images/rhpam-7-openshift-image/blob/3160e8313bbaa8c3777a2ac4fa5596e1849e7bf0/templates/rhpam74-kieserver-postgresql.yaml#L517)
> 5. **(IMPORTANT)** For all KIE Server DC, it requires this new downwardAPI mapping: (https://github.com/jboss-container-images/rhpam-7-openshift-image/blob/3160e8313bbaa8c3777a2ac4fa5596e1849e7bf0/templates/rhpam74-kieserver-postgresql.yaml#L618-L621)

Thanks for the detailed info, I just updated the PR addressing your points.

File: modules/rhpam-apb/roles/deploy-kieserver/tasks/routes.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@
       apiVersion: v1
       kind: Route
       metadata:
-        name: '{{ kieserver_deployment_name }}'
+        name: 'secure-{{ kieserver_deployment_name }}'
```
Comment (ruromero) [inline modules/rhpam-apb/roles/deploy-kieserver/tasks/routes.yml]: Why do you need to prefix it as `secure-` ? Is that related to the scripts?

Comment (rhtevan) [reply modules/rhpam-apb/roles/deploy-kieserver/tasks/routes.yml]: It is to be compatible with existing script. (https://github.com/jboss-container-images/jboss-kie-modules/blob/545387ac2073862138c6e46180d7dfe96bbeaf1f/jboss-kie-kieserver/added/launch/jboss-kie-kieserver.sh#L388)

It has been elaborated here.
https://issues.jboss.org/browse/KIECLOUD-97?focusedCommentId=13741158&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13741158

**Comment by spolti** [comment]
@errantepiphany @rhtevan updated and tested, ready to go.
